crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -1375,6 +1374,9 @@ impl PgManager {
         before: Option<String>,
     ) -> Result<Option<Connection<String, Coin>>, Error> {
         let address = address.into_vec();
+        let coin_type = coin_type.unwrap_or_else(|| {
+            GAS::type_().to_canonical_string(/* with_prefix */ true)
+        });
```

Comment:
Two thing to call out here:

- We also need to be canonicalizing types used as filters (this involves two steps:  One is to make the address lengths standard, and the other is to remap Package IDs to the defining IDs).
- This doesn't work with our usual testing instance, because the system types there have been indexed in their shortened form (`0x2::sui::SUI`) -- replacing this with `GAS::type_().to_string()` works for that, but I know we've already landed the canonicalization changes, so what's the best path forward for testing that? cc @wlmyng

Reply:
(Canonicalization and type package ID renormalization to be handled in a separate PR).

Reply:
> Two thing to call out here:
> 
> 
> 
> - We also need to be canonicalizing types used as filters (this involves two steps:  One is to make the address lengths standard, and the other is to remap Package IDs to the defining IDs).
> 
> - This doesn't work with our usual testing instance, because the system types there have been indexed in their shortened form (`0x2::sui::SUI`) -- replacing this with `GAS::type_().to_string()` works for that, but I know we've already landed the canonicalization changes, so what's the best path forward for testing that? cc @wlmyng 

Worth briefly discussing in our sync today?

Reply:
Yes, there's already topic in there about Prod DB schema changes